clippy_lints/src/derivable_impls.rs

```diff
@@ -128,11 +140,15 @@ fn check_struct<'tcx>(
         _ => false,
     };
 
+    let Some(derive_snippet) = determine_derive_macro(cx, is_const) else {
+        return;
+    };
+
     if should_emit {
```

Comment:
Best check for the derive macro to use only when we know that we will emit the lint:

```suggestion
    if should_emit && let Some(derive_snippet) = determine_derive_macro(cx, is_const) {
```